### PR TITLE
[Gutenberg] media upload progress reattach

### PR DESCRIPTION
**Pull request by mzorz**
#### Description
This PR introduces upload progress reattachment so the progress updates are reflected on the UI when you exit and enter the Editor again.

#### Related PRs
- [gutenberg-mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/505)
- [gutenberg PR](https://github.com/WordPress/gutenberg/pull/13516)

reattachprogress

#### To test:
1. start a new draft
2. insert an image
3. once it starts uploading, exit the editor
4. open the Post in the editor again
5. observe the upload progress bar is updated correspondingly.

#### Note:
@marecar3 the base branch corresponds to PR #9120, which also needs to be updated with latest `develop` to clear the commit history difference seen here.

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java [hunk 1]
```diff
@@ -283,7 +301,7 @@ public void setTitle(CharSequence title) {
             return;
         }
 
-        mWPAndroidGlueCode.setContent(title.toString(), null);
+        mWPAndroidGlueCode.setTitle(title.toString());
```
Comment (marecar3) [inline libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java]: Hmm not sure why this change is part of this PR?

File: libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java [hunk 2]
```diff
@@ -297,7 +315,7 @@ public void setContent(CharSequence text) {
         }
 
         String postContent = removeVisualEditorProgressTag(text.toString());
-        mWPAndroidGlueCode.setContent(null, postContent);
```
Comment (marecar3) [inline libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java]: Also for this one, not sure why is changed here?

File: libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java (hunk 1 above)
Comment (mzorz) [reply libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java]: I left a note in the description of the PR :D 
```
Note:
@marecar3 the base branch corresponds to PR #9120, which also needs to be updated with latest develop to clear the commit history difference seen here.
```

Comment (marecar3) [reply libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java]: Sorry and thanks! @mzorz

File: libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java [hunk 3]
```diff
@@ -58,6 +63,9 @@
 
     private WPAndroidGlueCode mWPAndroidGlueCode;
 
+    private ConcurrentHashMap<String, Float> mUploadingMediaProgressMax = new ConcurrentHashMap<>();
+    private Set<String> mFailedMediaIds = new HashSet<>();
```
Comment (marecar3) [inline libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java]: Any reason to use interface `Set` instead of `HashSet` ?

File: libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java [hunk 4]
```diff
@@ -112,11 +120,21 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+
                     @Override
                     public void onCapturePhotoButtonClicked() {
                         checkAndRequestCameraAndStoragePermissions();
                     }
                 },
+                new OnReattachQueryListener() {
+                    @Override
+                    public void onQueryCurrentProgressForUploadingMedia() {
+                        for (String mediaId : mUploadingMediaProgressMax.keySet()) {
```
Comment (marecar3) [inline libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java]: Hey @mzorz, what about moving this into a separate method (e.g. `updateMediaProgress` or similar) 
and also maybe we can put this `mUploadingMediaProgressMax.get(mediaId)` into variable (e.g. `int currentProgress`) before passing it to method ?

File: libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java (hunk 3 above)
Comment (mzorz) [reply libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java]: not really, just copied the code to represent the same functionality we had as parity with Aztec from here https://github.com/wordpress-mobile/WordPress-Android/blob/develop/libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java#L172

File: libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java (hunk 4 above)
Comment (mzorz) [reply libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java]: agreed, addressed moving into a new method in 6cb4c90456 - I think the intermediate local variable is unnecessary as it's only going to be used in 1 place; we can use that when it's referenced in more than 1 place so the `.get()` method doesn't get called on every reference. Does it make sense?
